pkg/actuators/machine/instances.go

```diff
@@ -134,3 +134,25 @@ func getNetworkID(network powervsproviderv1.PowerVSResourceReference, client pow
 
 	return nil, fmt.Errorf("failed to find a network ID")
 }
+
+func getServiceInstanceID(serviceInstance powervsproviderv1.PowerVSResourceReference, client powervsclient.Client) (*string, error) {
+	if serviceInstance.ID != nil {
+		klog.Infof("Found the service %v", serviceInstance)
+		return serviceInstance.ID, nil
+	} else if serviceInstance.Name != nil {
+		serviceInstances, err := client.GetCloudServiceInstances()
```

Comment:
currently this function call GetCloudServiceInstances returns only 100 instances by default and we need to have paging mechanism to handle the things properly, can you please open an issue to enhance this code?